test/metabase/sync/sync_metadata/fields_test.clj

```diff
@@ -47,127 +49,146 @@
     ;; ok, let's see what (f) gives us
     (let [f-before (f (data/db))]
       ;; ok cool! now delete one of those columns...
-      (jdbc/execute! one-off-dbs/*conn* ["ALTER TABLE \"birds\" DROP COLUMN \"example_name\";"])
+      (jdbc/execute! one-off-dbs/*conn* [alter-sql])
       ;; ...and re-sync...
       (sync/sync-database! (data/db))
       ;; ...now let's see how (f) may have changed! Compare to original.
       {:before-drop f-before
        :after-drop  (f (data/db))})))
 
+(deftest renaming-fields-test
+  (testing "make sure we can identify case changes on a field (#7923)"
+    (let [db-state (with-test-db-before-and-after-altering
+                     "ALTER TABLE \"birds\" RENAME COLUMN \"example_name\" to \"Example_Name\";"
+                     (fn [database]
+                       (set
+                        (map (partial into {})
+                             (db/select [Field :id :name :active]
+                                        :table_id [:in (db/select-ids Table :db_id (u/get-id database))])))))]
+      (is (= {:before-drop #{{:name "species",      :active true}
+                             {:name "example_name", :active true}}
+              :after-drop #{{:name "species",      :active true}
```

Comment:
Minor: Maybe `:before-sync` and `:after-sync` are better names?

Reply:
good catch. done!